### PR TITLE
OPENEUROPA-3114: Add coordinators and participants fields to project content type.

### PR DESCRIPTION
**Pull request by junkuncz**

### CODE REVIEW

File: tests/Behat/FeatureContext.php [hunk 1]
```diff
@@ -298,4 +299,20 @@ public function assertVisibility($element): void {
     $this->assertVisuallyVisible($node);
   }
 
+  /**
+   * Create a dummy stakeholder.
```
Comment (drishu) [inline tests/Behat/FeatureContext.php:303]: drop dummy

File: tests/Behat/FeatureContext.php [hunk 2]
```diff
@@ -298,4 +299,20 @@ public function assertVisibility($element): void {
...
+   *
+   * @todo Make it more flexible.
```
Comment (drishu) [inline tests/Behat/FeatureContext.php:305]: drop this comment, and add example usage

File: tests/Behat/FeatureContext.php [hunk 3]
```diff
@@ -298,4 +299,20 @@ public function assertVisibility($element): void {
...
+   *
+   * @Given a dummy :name stakeholder
```
Comment (drishu) [inline tests/Behat/FeatureContext.php:307]: drop dummy

File: tests/Behat/FeatureContext.php [hunk 4]
```diff
@@ -298,4 +299,20 @@ public function assertVisibility($element): void {
...
+   */
+  public function givenStakeholder($name) {
```
Comment (drishu) [inline tests/Behat/FeatureContext.php:309]: use TableNode as parameter

File: tests/Behat/FeatureContext.php [hunk 5]
```diff
@@ -298,4 +299,20 @@ public function assertVisibility($element): void {
...
+    $stakeholder = Organisation::create([
```
Comment (drishu) [inline tests/Behat/FeatureContext.php:310]: dont use variable call save directly on the return

File: tests/features/project.feature [hunk 6]
```diff
@@ -7,6 +7,8 @@ Feature: Project content creation
   @javascript
   Scenario: Creation of a Project content through the UI.
     Given I am logged in as a user with the "create oe_project content, access content, edit own oe_project content, view published skos concept entities, manage corporate content entities" permission
+    And a dummy "My stakeholder 1" stakeholder
```
Comment (drishu) [inline tests/features/project.feature:10]: with TableNode you will call this in a table like below for documents instead of duplicate lines

File: tests/features/project.feature [hunk 7]
```diff
@@ -40,10 +44,14 @@ Feature: Project content creation
 
     # Add link fields.
     When I press "Save"
+
+    # @todo: actually it should be fine, but it's failing now.
```
Comment (drishu) [inline tests/features/project.feature:48]: remove this

File: tests/Behat/FeatureContext.php [hunk 8]
```diff
@@ -298,4 +298,30 @@ public function assertVisibility($element): void {
...
+    foreach ($stakeholders as $stakeholder) {
+      $entity = $storage->create([
+        'bundle' => 'oe_stakeholder',
+        'name' => $stakeholder['name'],
+        'status' => 1,
+      ]);
+      $entity->save();
+    }
+  }
+
```
Comment (ademarco) [inline tests/Behat/FeatureContext.php:326]: We need to provide a corporate reusable context, check what we did here:
```
./tests/Behat/Content/Venue/DefaultVenueContext.php
./tests/Behat/Content/Contact/PressContactContext.php
./tests/Behat/Content/Contact/GeneralContactContext.php
```

Comment (drishu) [reply tests/Behat/FeatureContext.php]: refactored

Comment (ademarco) [reply tests/Behat/FeatureContext.php]: Sorry @drishu, what I meant was to add a specific context for the stakeholder organisation bundle, like those 3 above, so something named like:

```
./tests/Behat/Content/Organisation/StakeholderOrganisationContext.php
```
Then in the `ProjectContentContext` you created we reference those entities.

Check what we did for event and contact entities.

File: tests/Behat/Content/Node/ProjectContentContext.php [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+   * Creates stakeholder/s for organisation entity.
+   *
+   * Usage example:
+   *
+   * Given the following stakeholders:
+   *   | name   |
+   *   | name 2 |
+   *   |   ...  |
+   *
+   * @Given the following stakeholder(s):
```
Comment (ademarco) [inline tests/Behat/Content/Node/ProjectContentContext.php:25]: We don't have to use this at all, we need to use our Behat APIs instead.

File: modules/oe_content_entity/tests/src/FunctionalJavascript/StakeholderUiTest.php [hunk 10]
```diff
@@ -0,0 +1,88 @@
+<?php
```
[the remark this answers is not in the document]
Comment (julien-) [reply modules/oe_content_entity/tests/src/FunctionalJavascript/StakeholderUiTest.php]: Previously fixed

File: tests/Behat/FeatureContext.php (hunk 1 above)
Comment (julien-) [reply tests/Behat/FeatureContext.php]: Previously fixed

File: tests/Behat/FeatureContext.php (hunk 2 above)
Comment (julien-) [reply tests/Behat/FeatureContext.php]: Previously fixed

File: tests/Behat/FeatureContext.php (hunk 3 above)
Comment (julien-) [reply tests/Behat/FeatureContext.php]: Previously fixed

File: tests/Behat/FeatureContext.php (hunk 4 above)
Comment (julien-) [reply tests/Behat/FeatureContext.php]: Previously fixed

File: tests/Behat/FeatureContext.php (hunk 5 above)
Comment (julien-) [reply tests/Behat/FeatureContext.php]: Previously fixed

File: tests/features/project.feature (hunk 6 above)
Comment (julien-) [reply tests/features/project.feature]: Previously fixed

File: tests/features/project.feature (hunk 7 above)
Comment (julien-) [reply tests/features/project.feature]: Previously fixed

File: tests/Behat/FeatureContext.php (hunk 8 above)
Comment (julien-) [reply tests/Behat/FeatureContext.php]: Is it possible to add a commun function to manage same behavior and not copy paste the same code :

```
foreach ($scope->getFields() as $key => $value) {

// Handle entity references.
switch ($key) {
  case 'Published':
    $scope->addFields([
    $mapping[$key] => (int) ($value === 'Yes'),
  ])->removeField($key);
  break;

  default:
    if (isset($mapping[$key])) {
      $scope->renameField($key, $mapping[$key]);
    }
  }
}
```
?


For that, I suggest to add the following function in `tests/Behat/Content/CorporateContentContext.php` :

```
/**
 * Run before fields are parsed by Drupal Behat extension.
 *
 * @param \Drupal\Tests\oe_content\Behat\Hook\Scope\BeforeParseEntityFieldsScope $scope
 *   Behat hook scope.
 *
 * @BeforeParseEntityFields(oe_organisation,oe_stakeholder)
 */
public function mapFieldsLabelName(BeforeParseEntityFieldsScope $scope): void {
  // Maps human readable field names to their Behat parsable machine names.
  $bundle_fields = \Drupal::getContainer()->get('entity_field.manager')->getFieldDefinitions($scope->getEntityType(), $scope->getBundle());
  // var_dump(array_keys($bundle_fields));
  $mapping = [];
  foreach($bundle_fields as $field_name => $field_config){
    if(is_string($label = $field_config->getLabel())){
    $mapping[$label] = $field_name;
    }
  }
  $mapping['Published'] = 'status';
  
  foreach ($scope->getFields() as $key => $value) {
    switch ($key) {
      case 'Published':
        $scope->addFields([
          $mapping[$key] => (int) ($value === 'Yes'),
        ])->removeField($key);
        break;

      default:
        if (isset($mapping[$key])) {
          $scope->renameField($key, $mapping[$key]);
        }
    }
  }
}
```

File: modules/oe_content_project/config/install/field.storage.node.oe_project_coordinators.yml [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+id: node.oe_project_coordinators
+field_name: oe_project_coordinators
+entity_type: node
+type: entity_reference_revisions
+settings:
+  target_type: oe_organisation
+module: core
+locked: false
+cardinality: -1
+translatable: true
```
Comment (nagyad) [inline modules/oe_content_project/config/install/field.storage.node.oe_project_coordinators.yml:17]: Should not be translatable.

File: modules/oe_content_project/config/install/field.storage.node.oe_project_participants.yml [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+id: node.oe_project_participants
+field_name: oe_project_participants
+entity_type: node
+type: entity_reference_revisions
+settings:
+  target_type: oe_organisation
+module: core
+locked: false
+cardinality: -1
+translatable: true
```
Comment (nagyad) [inline modules/oe_content_project/config/install/field.storage.node.oe_project_participants.yml:17]: Should not be translatable.

File: tests/Behat/Content/Node/ProjectContentContext.php [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+      'Teaser' => 'oe_teaser',
+      'Departments' => 'oe_project_departments',
+      'Coordinators' => 'oe_project_coordinators',
+      'Funding programme' => 'oe_project_funding_programme',
+      'Participants' => 'oe_project_participants',
+      'Published' => 'status',
+    ];
+
+    foreach ($scope->getFields() as $key => $value) {
+
```
Comment (nagyad) [inline tests/Behat/Content/Node/ProjectContentContext.php:48]: This empty line is not required.

File: tests/features/project.feature [hunk 14]
```diff
@@ -1,4 +1,4 @@
-@api
+@api @project
```
Comment (nagyad) [inline tests/features/project.feature:1]: Is this tag has any related function? I can't see any.

File: tests/features/project.feature [hunk 15]
```diff
@@ -7,17 +7,26 @@ Feature: Project content creation
   @javascript
   Scenario: Creation of a Project content through the UI.
     Given I am logged in as a user with the "create oe_project content, access content, edit own oe_project content, view published skos concept entities, manage corporate content entities" permission
+    # Create Stakeholders.
+    Given the following Stakeholder Organisation entity:
+    | Name       | Coordinators stakeholder |
+    | Published  | Yes                      |
```
Comment (nagyad) [inline tests/features/project.feature:13]: Wrong indentation.

File: tests/features/project.feature [hunk 16]
```diff
@@ -7,17 +7,26 @@ Feature: Project content creation
   @javascript
   Scenario: Creation of a Project content through the UI.
     Given I am logged in as a user with the "create oe_project content, access content, edit own oe_project content, view published skos concept entities, manage corporate content entities" permission
+    # Create Stakeholders.
+    Given the following Stakeholder Organisation entity:
```
Comment (nagyad) [inline tests/features/project.feature:11]: And, you have a Given keyword already.

File: tests/features/project.feature [hunk 17]
```diff
@@ -7,17 +7,26 @@ Feature: Project content creation
...
+    | Name       | Coordinators stakeholder |
+    | Published  | Yes                      |
+    Given the following Stakeholder Organisation entity:
```
Comment (nagyad) [inline tests/features/project.feature:14]: And

File: tests/features/project.feature [hunk 18]
```diff
@@ -7,17 +7,26 @@ Feature: Project content creation
...
+    | Name      | Participants stakeholder  |
+    | Published | Yes                       |
```
Comment (nagyad) [inline tests/features/project.feature:16]: Wrong indentation.

File: modules/oe_content_project/config/install/field.storage.node.oe_project_coordinators.yml (hunk 11 above)
Comment (drishu) [reply modules/oe_content_project/config/install/field.storage.node.oe_project_coordinators.yml]: dono how I missed this

File: modules/oe_content_project/config/install/field.storage.node.oe_project_participants.yml (hunk 12 above)
Comment (drishu) [reply modules/oe_content_project/config/install/field.storage.node.oe_project_participants.yml]: fixed

File: tests/Behat/Content/Node/ProjectContentContext.php (hunk 13 above)
Comment (drishu) [reply tests/Behat/Content/Node/ProjectContentContext.php]: removed

File: tests/features/project.feature (hunk 14 above)
Comment (drishu) [reply tests/features/project.feature]: none, removed

File: tests/features/project.feature (hunk 15 above)
Comment (drishu) [reply tests/features/project.feature]: fixed

File: tests/features/project.feature (hunk 16 above)
Comment (drishu) [reply tests/features/project.feature]: fixed

File: tests/features/project.feature (hunk 17 above)
Comment (drishu) [reply tests/features/project.feature]: fixed

File: tests/features/project.feature (hunk 18 above)
Comment (drishu) [reply tests/features/project.feature]: fixed

File: tests/Behat/Content/Node/ProjectContentContext.php [hunk 19]
```diff
@@ -0,0 +1,71 @@
...
+      'Teaser' => 'oe_teaser',
+      'Departments' => 'oe_project_departments',
+      'Coordinators' => 'oe_project_coordinators',
+      'Funding programme' => 'oe_project_funding_programme',
+      'Participants' => 'oe_project_participants',
+      'Published' => 'status',
+    ];
+
+    foreach ($scope->getFields() as $key => $value) {
+      // Handle entity references.
```
Comment (nagyad) [inline tests/Behat/Content/Node/ProjectContentContext.php:48]: This comment is misleading here, you are not handling entity references.

File: tests/Behat/Content/Node/ProjectContentContext.php [hunk 20]
```diff
@@ -0,0 +1,70 @@
...
+      'Results' => 'oe_project_results',
+      'Result files' => 'oe_project_result_files',
+      'Alternative title' => 'oe_content_short_title',
+      'Navigation title' => 'oe_content_navigation_title',
+      'Teaser' => 'oe_teaser',
+      'Departments' => 'oe_project_departments',
+      'Coordinators' => 'oe_project_coordinators',
+      'Funding programme' => 'oe_project_funding_programme',
+      'Participants' => 'oe_project_participants',
+      'Published' => 'status',
```
Comment (ademarco) [inline tests/Behat/Content/Node/ProjectContentContext.php:44]: Many of this fields will need further processing in order for them to be usable in a Behat test, such as `oe_project_departments` that references a SKOS vocabulary.

For fields such as that, on the event context, we have something like:

```php
        case 'Type':
        case 'Languages':
        case 'Internal organiser':
          $fields = $this->getRevisionField($mapping[$key], 'skos_concept', $value);
          $scope->addFields($fields)->removeField($key);
          break;
```

So we either complete each one of the fields above properly, by also adding a test that proves that this works, or we only add here the minimum, such as "title", "published" and other simple value fields (text, float, long text, etc.).

Or we can simply remove this context all together, as it's not used in this PR. We could move this in the theme ticket, as a companion PR on the oe_content, as this context will be used to test the project theming.

Up to you, only if we ship with this, then let's make sure that what we ship is complete and working properly, albeit minimal.

File: tests/features/project.feature [hunk 21]
```diff
@@ -7,17 +7,26 @@ Feature: Project content creation
   @javascript
   Scenario: Creation of a Project content through the UI.
     Given I am logged in as a user with the "create oe_project content, access content, edit own oe_project content, view published skos concept entities, manage corporate content entities" permission
+    # Create Stakeholders.
+    And the following Stakeholder Organisation entity:
+      | Name       | Coordinators stakeholder |
+      | Published  | Yes                      |
+    And the following Stakeholder Organisation entity:
+      | Name      | Participants stakeholder  |
+      | Published | Yes                       |
```
Comment (ademarco) [inline tests/features/project.feature:16]: Let's use here all fields that are parsed in `alterStakeholderOrganisationFields` so that we also test that they work properly.

Comment (drishu) [reply tests/features/project.feature]: done

File: tests/Behat/Content/Node/ProjectContentContext.php (hunk 20 above)
Comment (drishu) [reply tests/Behat/Content/Node/ProjectContentContext.php:46]: done

File: tests/Behat/Content/Node/ProjectContentContext.php [hunk 22]
```diff
@@ -4,38 +4,96 @@
...
+          }
+      }
     }
+
+    // Set default fields.
+    $scope->addFields([
+      'oe_subject' => 'http://data.europa.eu/uxp/10',
+      'oe_author' => 'http://publications.europa.eu/resource/authority/corporate-body/AASM',
+      'oe_content_content_owner' => 'http://publications.europa.eu/resource/authority/corporate-body/AASM',
+    ]);
```
[the remark this answers is not in the document]
Comment (drishu) [reply tests/Behat/Content/Node/ProjectContentContext.php:96]: we will use in oe_theme

File: tests/features/project.feature [hunk 23]
```diff
@@ -7,12 +7,40 @@ Feature: Project content creation
...
+    And the following Stakeholder Organisation entity:
+      | Name                      | Coordinators stakeholder  |
+      | Acronym                   | CS                        |
+      | oe_address:country_code   | FR                        |
+      |           :organization   | Ma société 1              |
+      |           :address_line1  | 1 rue de la Paix          |
+      |           :address_line2  | Etage 1                   |
+      |           :postal_code    | 11111                     |
+      |           :locality       | Cruscades                 |
+      |           :code           | 1                         |
```
[the remark this answers is not in the document]
Comment (julien-) [reply tests/features/project.feature]: rewrited

File: tests/features/project.feature [hunk 24]
```diff
@@ -7,12 +7,40 @@ Feature: Project content creation
...
+      | Published         | Yes                               |
+    And the following Stakeholder Organisation entity:
+      | Name                      | Participants stakeholder  |
+      | Acronym                   | PS                        |
+      | oe_address:country_code   | LU                        |
+      |           :organization   | Ma société 2              |
+      |           :address_line1  | 2 Avenue du Blues         |
+      |           :address_line2  | Etage 2                   |
+      |           :postal_code    | 2222                      |
+      |           :locality       | Luxembourg city           |
```
[the remark this answers is not in the document]
Comment (julien-) [reply tests/features/project.feature]: rewrited

File: tests/features/project.feature [hunk 25]
```diff
@@ -7,12 +7,40 @@ Feature: Project content creation
...
+      | Contact page URL  | https://ec.europa.eu/contact2     |
+      | Logo              | Image 2                           |
+      | Website           | https://ec.europa.eu/2            |
+      | Published         | Yes                               |
```
[the remark this answers is not in the document]
Comment (julien-) [reply tests/features/project.feature]: Block reformated

File: tests/features/project.feature [hunk 26]
```diff
@@ -7,12 +7,40 @@ Feature: Project content creation
...
+      |           :organization   | Ma société 1              |
+      |           :address_line1  | 1 rue de la Paix          |
+      |           :address_line2  | Etage 1                   |
+      |           :postal_code    | 11111                     |
+      |           :locality       | Cruscades                 |
+      |           :code           | 1                         |
+      | Contact page URL  | https://ec.europa.eu/contact1     |
+      | Logo              | Image 1                           |
+      | Website           | https://ec.europa.eu/1            |
+      | Published         | Yes                               |
```
[the remark this answers is not in the document]
Comment (julien-) [reply tests/features/project.feature]: Block reformated

File: tests/Behat/Content/Node/ProjectContentContext.php [hunk 27]
```diff
@@ -0,0 +1,99 @@
...
+          $scope->addFields($fields)->removeField($key);
+          break;
+
+        case 'Coordinators':
+        case 'Participants':
+          $fields = $this->getReferenceRevisionField($mapping[$key], 'oe_organisation', $value);
+          $scope->addFields($fields)->removeField($key);
+          break;
+
+        // Custom case.
```
Comment (ademarco) [inline tests/Behat/Content/Node/ProjectContentContext.php:77]: Use a more descriptive comment, if any, like "Set content published status."

File: tests/features/project.feature [hunk 28]
```diff
@@ -38,18 +61,51 @@ Feature: Project content creation
...
+    And I should see "Participants stakeholder"
+    And I should see "PS"
+    And I should see "Luxembourg"
+    And I should see "Ma société 2"
+    And I should see "2 Avenue du Blues"
+    And I should see "Etage 2"
+    And I should see "2222"
+    And I should see "Luxembourg city"
+    And I should see "https://ec.europa.eu/contact2"
+    And I should see "Logo"
```
Comment (ademarco) [inline tests/features/project.feature:93]: This step is already present on the top portion, so we have two times 
```
     And I should see "Logo"
...
     And I should see "Logo"
```
And we don't know to which logo we are referring. I'd suggest to add a dedicated Behat region representing each field, and then we can assert: 
```gherkin
     And I should see "Logo" in the "Project participants" region
```
Same for the coordinators.

File: tests/features/project.feature [hunk 29]
```diff
@@ -38,18 +61,51 @@ Feature: Project content creation
...
+    And I should see "2019-02-24"
+    And I should see "1000"
+    And I should see "200"
+    And I should see "Anti Fraud Information System (AFIS)"
     And I should see "Website"
     And I should see "Example proposal"
+    And I should see "Result 1 text"
+    And I should see "sample.pdf"
+    And I should not see "My alternative title text"
+    And I should not see "My navigation title text"
```
Comment (ademarco) [inline tests/features/project.feature:110]: Let's remove these two negative assertions, as business-wise they have no value here.

File: tests/Behat/Content/Node/ProjectContentContext.php [hunk 30]
```diff
@@ -0,0 +1,99 @@
...
+      'Results' => 'oe_project_results',
+      'Summary' => 'oe_summary',
+      'Teaser' => 'oe_teaser',
+      'Title' => 'title',
+      'Website' => 'oe_project_website',
+    ];
+
+    foreach ($scope->getFields() as $key => $value) {
+      switch ($key) {
+        // Type skos_concept_entity_reference.
```
[the remark this answers is not in the document]
Comment (drishu) [reply tests/Behat/Content/Node/ProjectContentContext.php:58]: done

File: tests/Behat/Content/Node/ProjectContentContext.php [hunk 31]
```diff
@@ -0,0 +1,99 @@
...
+        case 'Departments':
+        case 'Funding programme':
+          $fields = $this->getRevisionField($mapping[$key], 'skos_concept', $value);
+          $scope->addFields($fields)->removeField($key);
+          break;
+
+        // Type entity_reference_revisions.
```
[the remark this answers is not in the document]
Comment (drishu) [reply tests/Behat/Content/Node/ProjectContentContext.php:65]: done

File: tests/Behat/Content/Node/ProjectContentContext.php (hunk 27 above)
Comment (drishu) [reply tests/Behat/Content/Node/ProjectContentContext.php:77]: done

File: tests/features/project.feature (hunk 28 above)
Comment (drishu) [reply tests/features/project.feature:93]: done

File: tests/features/project.feature (hunk 29 above)
Comment (drishu) [reply tests/features/project.feature:110]: done

File: tests/features/project.feature [hunk 32]
```diff
@@ -39,17 +62,48 @@ Feature: Project content creation
...
+    And I should see "France"
+    And I should see "Ma société 1"
+    And I should see "1 rue de la Paix"
+    And I should see "Etage 1"
+    And I should see "11111"
+    And I should see "Cruscades"
+    And I should see "1"
+    And I should see "https://ec.europa.eu/contact1"
+    Then I should see "Logo" in the "Project coordinators" region
+    And I should see "https://ec.europa.eu/1"
```
Comment (ademarco) [inline tests/features/project.feature:82]: @drishu sorry, I mean to add ` in the "Project coordinators" region` to all the assertions here above, same for the Participants

File: tests/features/project.feature (hunk 29 above)
Comment (ademarco) [reply tests/features/project.feature:108]: I still see these two assertions.

File: tests/features/project.feature (hunk 32 above)
Comment (enriquelacoma) [reply tests/features/project.feature]: Added regions to assertions

File: tests/features/project.feature [hunk 33]
```diff
@@ -46,19 +62,46 @@ Feature: Project content creation
...
+    And I should see "CS" in the "Project coordinators" region
+    And I should see "France" in the "Project coordinators" region
+    And I should see "Ma société 1" in the "Project coordinators" region
+    And I should see "1 rue de la Paix" in the "Project coordinators" region
+    And I should see "Etage 1" in the "Project coordinators" region
+    And I should see "11111" in the "Project coordinators" region
+    And I should see "Cruscades" in the "Project coordinators" region
+    And I should see "1" in the "Project coordinators" region
+    And I should see "https://ec.europa.eu/contact1" in the "Project coordinators" region
+    Then I should see "Logo" in the "Project coordinators" region
```
Comment (ademarco) [inline tests/features/project.feature:81]: `And`

File: tests/features/project.feature [hunk 34]
```diff
@@ -46,19 +62,46 @@ Feature: Project content creation
...
+    And I should see "Participants stakeholder" in the "Project participants" region
+    And I should see "PS" in the "Project participants" region
+    And I should see "Luxembourg" in the "Project participants" region
+    And I should see "Ma société 2" in the "Project participants" region
+    And I should see "2 Avenue du Blues" in the "Project participants" region
+    And I should see "Etage 2" in the "Project participants" region
+    And I should see "2222" in the "Project participants" region
+    And I should see "Luxembourg city" in the "Project participants" region
+    And I should see "https://ec.europa.eu/contact2" in the "Project participants" region
+    Then I should see "Logo" in the "Project participants" region
```
Comment (ademarco) [inline tests/features/project.feature:93]: `And`

File: tests/features/project.feature (hunk 33 above)
Comment (enriquelacoma) [reply tests/features/project.feature]: Changed Then to And

File: tests/features/project.feature (hunk 34 above)
Comment (enriquelacoma) [reply tests/features/project.feature]: Changed Then to And

File: tests/Behat/Content/Node/ProjectContentContext.php [hunk 35]
```diff
@@ -0,0 +1,99 @@
...
+      'Results' => 'oe_project_results',
+      'Summary' => 'oe_summary',
+      'Teaser' => 'oe_teaser',
+      'Title' => 'title',
+      'Website' => 'oe_project_website',
+    ];
+
+    foreach ($scope->getFields() as $key => $value) {
+      switch ($key) {
+        // Set SKOS Concemt entity reference fields.
```
[the remark this answers is not in the document]
Comment (drishu) [reply tests/Behat/Content/Node/ProjectContentContext.php]: fixed

File: tests/features/project.feature [hunk 36]
```diff
@@ -7,29 +7,52 @@ Feature: Project content creation
...
     And I fill in "Subject" with "EU financing"
     And I fill in "Author" with "European Patent Office"
     And I fill in "Body text" with "Body text"
+    And I fill in "Coordinators" with "Coordinators stakeholder"
+    And I fill in "Participants" with "Participants stakeholder"
     And I fill in "Teaser" with "Project teaser text" in the "Alternative titles and teaser" region
     And I fill in "Content owner" with "Committee on Agriculture and Rural Development"
-
-    # Optional fields.
+    # Fill up optional fields.
```
[the remark this answers is not in the document]
Comment (drishu) [reply tests/features/project.feature]: fixed

File: tests/features/project.feature [hunk 37]
```diff
@@ -7,29 +7,52 @@ Feature: Project content creation
...
+      | Published         | Yes                                                                                                                                                       |
+
+    # Create a document, to be referenced later on.
+    And the following document:
+      | name          | file       |
+      | My Document 1 | sample.pdf |
 
     When I visit "the Project creation page"
-    # Mandatory fields.
+    # Fill up mandatory fields.
```
[the remark this answers is not in the document]
Comment (drishu) [reply tests/features/project.feature]: fixed
